cpp/src/io/utilities/wrapper_utils.hpp

```diff
@@ -146,3 +146,39 @@ struct rmm_deleter {
 };
 template <typename T>
 using device_ptr = std::unique_ptr<T, rmm_deleter<T>>;
+
+// DOXY
+template <typename T>
+class rmm_device_buffer {
+	T* d_data = nullptr;
+
+public:
+	rmm_device_buffer() noexcept = default;
+	rmm_device_buffer(size_t cnt) {
+		const auto error = RMM_ALLOC(&d_data, cnt*sizeof(T), 0);
+		if(error != RMM_SUCCESS) {
+			cudf::detail::throw_cuda_error(cudaErrorMemoryAllocation, __FILE__, __LINE__);
+		}
+	}
+
+	T* data() const noexcept {return d_data;}
+
+	void resize(size_t cnt) {
+		const auto error = RMM_REALLOC(&d_data, cnt*sizeof(T), 0);
+		if(error != RMM_SUCCESS) {
+			cudf::detail::throw_cuda_error(cudaErrorMemoryAllocation, __FILE__, __LINE__);
+		}
+	}
+	// TODO: do we want to enable copy assignment?
+	rmm_device_buffer& operator=(rmm_device_buffer& ) = delete;
```

Comment:
I would delete the copy constructor as well.

Reply:
The copy constructor is implicitly deleted because the destructor is declared.
I'll delete it explicitly to make the intent clear.

Reply:
Not quite that simple. You are correct that the implicit copy constructor here is deleted, but that's because you have a user-defined move constructor.

See:
https://wandbox.org/permlink/4Aww7zewZMSf69qF
https://en.cppreference.com/w/cpp/language/copy_constructor#Deleted_implicitly-declared_copy_constructor

Reply:
Darn, mixed up the rules. Illustrates why it's important to delete the special functions explicitly :)